In [21]:
import os
import pandas as pd

In [22]:
def myCheckFiles(csv_file_path, image_folder_path):
    # Đọc tên ảnh từ file CSV
    csv_data = pd.read_csv(csv_file_path)
    image_numbers_from_csv = csv_data['Picture Name'].tolist()  # Thay 'image_name_column' bằng tên cột chứa tên ảnh

    # Chuyển đổi số ảnh từ file CSV thành danh sách số
    image_numbers_from_csv = [int(num) for num in image_numbers_from_csv]

    # Lấy danh sách tên ảnh từ thư mục chứa ảnh
    image_names_from_folder = os.listdir(image_folder_path)

    # Tách phần số từ tên ảnh trong thư mục chứa ảnh
    image_numbers_from_folder = [int(name.split('.')[0]) for name in image_names_from_folder]

    # Tìm các ảnh có trong thư mục nhưng không có trong file CSV
    missing_in_csv = set(image_numbers_from_folder) - set(image_numbers_from_csv)

    # Tìm các ảnh có trong file CSV nhưng không có trong thư mục ảnh
    extra_in_csv = set(image_numbers_from_csv) - set(image_numbers_from_folder)

    # In ra các kết quả
    print("Các ảnh có trong thư mục nhưng không có trong file CSV:", missing_in_csv)
    print("Các ảnh có trong file CSV nhưng không có trong thư mục ảnh:", extra_in_csv)

In [23]:
# Đường dẫn tới file CSV và thư mục chứa ảnh
csv_file_path = 'path/to/your/csv/file' #(dataset.csv)
image_folder_path = 'path/to/your/folder/image'

myCheckFiles(csv_file_path, image_folder_path)

Các ảnh có trong thư mục nhưng không có trong file CSV: {6514, 2211, 8509, 6515}
Các ảnh có trong file CSV nhưng không có trong thư mục ảnh: set()


- Như vậy có 4 ảnh thiếu trong file CSV, nhưng ta có 1118 dòng trong file csv và 1120 ảnh trong folder.
- Có thể là trong csv có dòng bị lặp.

In [24]:
# Kiểm tra các dòng bị lặp
# Đọc file CSV vào DataFrame
df = pd.read_csv(csv_file_path)

# Kiểm tra các dòng bị trùng lặp dựa trên một cột cụ thể
duplicated_rows = df[df.duplicated(subset=['Picture Name'])]  # Thay 'image_name_column' bằng tên cột cụ thể

# In ra các dòng bị trùng lặp
if not duplicated_rows.empty:
    print("Các dòng bị trùng lặp dựa trên cột cụ thể:")
    print(duplicated_rows["Picture Name"])
else:
    print("Không có dòng nào bị trùng lặp trong file CSV dựa trên cột cụ thể.")

Các dòng bị trùng lặp dựa trên cột cụ thể:
207    2206
792    8506
Name: Picture Name, dtype: int64


- Vậy là ta biết được các dòng bị lặp, thực hiện sửa và đối chiếu với các ảnh trong folder
- Mở file dataset.csv lên ta thấy dòng 208 và 209 lặp lại => Đối chiếu với các ảnh trong folder, ta nhận thấy ảnh 2209, 2210 và 2211 mới là ảnh NIGHT => Sửa từ dòng 209 đến dòng 213: từ 2207 -> 2211.
- Và dòng 791, 794 là lặp lại. Ta chỉnh dòng 791 -> 793, (tương tự xem cột NIGHT đối chiếu ảnh) sửa thành 8507, 8508, 8509. Dòng 794 là đúng vì DAY, giữ nguyên.

In [25]:
# Đường dẫn tới file CSV đã sửa
csv_modified_file_path = 'path/to/modified/csv/file' #(dataset_modified.csv)

myCheckFiles(csv_modified_file_path, image_folder_path)

Các ảnh có trong thư mục nhưng không có trong file CSV: {6514, 6515}
Các ảnh có trong file CSV nhưng không có trong thư mục ảnh: set()


- Sau khi sửa xong thì ta vẫn thấy có 2 ảnh không cần sử dụng là 6514, 6515, thực hiện chuyển sang folder ảnh không có trong csv.